### PR TITLE
History cleanup scanner for Cassandra

### PR DESCRIPTION
**Pull request by longquanzheng**
some logic is copied from https://github.com/uber/cadence/blob/master/service/worker/batcher/workflow.go

### CODE REVIEW

File: common/persistence/dataInterfaces.go [hunk 1]
```diff
@@ -2392,6 +2393,20 @@ func NewHistoryBranchToken(treeID string) ([]byte, error) {
 	return token, nil
 }
 
+// NewHistoryBranchTokenByBranchID return a new branch token with treeID/branchID
+func NewHistoryBranchTokenByBranchID(treeID, branchID string) ([]byte, error) {
```
Comment (yux0) [inline common/persistence/dataInterfaces.go]: nit: looks like we are adopt the format like each parameter takes a new line.

File: common/persistence/dataInterfaces.go [hunk 2]
```diff
@@ -2411,3 +2426,15 @@ func NewHistoryBranchTokenFromAnother(branchID string, anotherToken []byte) ([]b
 	}
 	return token, nil
 }
+
+func BuildHistoryGarbageCleanupInfo(domainID, workflowID, runID string) string {
+	return fmt.Sprintf("%v:%v:%v", domainID, workflowID, runID)
+}
+
+func SplitHistoryGarbageCleanupInfo(info string) (domainID, workflowID, runID string, err error) {
+	ss := strings.Split(info, ":")
```
Comment (yux0) [inline common/persistence/dataInterfaces.go]: Does workflow id allow colon?

File: service/worker/scanner/history/scavenger.go [hunk 3]
```diff
@@ -0,0 +1,304 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
```
Comment (yux0) [inline service/worker/scanner/history/scavenger.go]: 2019

File: common/persistence/dataInterfaces.go [hunk 4]
```diff
@@ -2411,3 +2426,15 @@ func NewHistoryBranchTokenFromAnother(branchID string, anotherToken []byte) ([]b
...
+	if len(ss) != 3 {
```
Comment (wxing1292) [inline common/persistence/dataInterfaces.go]: plz make the `3` a constant

File: service/worker/scanner/history/scavenger.go [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+		}
+
+		if len(s.hbd.NextPageToken) == 0 {
+			break
+		}
+	}
+	return s.hbd, nil
+}
+
+func (s *Scavenger) startTaskProcessor(ctx context.Context, taskCh chan taskDetail, respCh chan error) {
```
Comment (wxing1292) [inline service/worker/scanner/history/scavenger.go]: plz do an indentation here

**Pull request merged**